iwyu.cc

```diff
@@ -3102,7 +3102,8 @@ class IwyuBaseAstVisitor : public BaseAstVisitor<Derived> {
 
     // If we are forward-declarable, so are our template arguments.
     if (CanForwardDeclareType(current_ast_node())) {
-      ReportDeclForwardDeclareUse(CurrentLoc(), decl);
+      if (!current_ast_node()->template GetAncestorAs<Decl>()->isImplicit())
```

Comment:
This construct is used now in 3 places. Maybe, it should be a function `InImplicitCode`?